tools/util.h

```diff
@@ -1,5 +1,12 @@
-#ifndef _UTIL_H_
-#define _UTIL_H_
+#ifndef UTIL_H
+#define UTIL_H
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
```

Comment:
stddef for the `size_t`s, `stdint` for the `uint32_t`s and `uint8_t`s, `stdio` for `printf`. I was getting warnings when compiling without (and missing struct members in other headers in VSCode), that's why I added them to the other headers.

Reply:
shouldn't the files using printf include stdio themselves? sounds like a bit of a heavy include for a util header

Reply:
But the header itself references it:
https://github.com/zeldaret/oot/blob/73d4ed5af2b61a9affb6cd4d4c158da7cc6f5b1a/tools/util.h#L11-L13

Reply:
I missed that 😅
And I also have no clue what it's for
Whatever x2

Reply:
The `format` attribute informs gcc (and other supporting compilers) that `util_fatal_error` is going to use argument `1` as a `printf` format string populated by the variadic arguments starting from argument `2`. This lets it do compile time type checks by examining the format specifiers.
The `noreturn` attribute lets the compiler possibly perform optimizations that would otherwise be unsafe if the function returned.
(https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#Common-Function-Attributes)